app/actions/remote/entry/gql_common.ts

```diff
@@ -98,6 +102,16 @@ export async function deferredAppEntryGraphQLActions(
     updateCanJoinTeams(serverUrl);
     updateAllUsersSince(serverUrl, since);
 
+    // defer sidebar GM profiles
+    setTimeout(async () => {
+        const directChannels = chData?.channels?.filter((v) => v?.type === General.GM_CHANNEL);
+        const channelsToFetchProfiles = new Set<Channel>(directChannels);
+        if (channelsToFetchProfiles.size) {
+            const teammateDisplayNameSetting = getTeammateNameDisplaySetting(preferences || [], config.LockTeammateNameDisplay, config.TeammateNameDisplay, license);
+            fetchMissingDirectChannelsInfo(serverUrl, Array.from(channelsToFetchProfiles), currentUserLocale, teammateDisplayNameSetting, currentUserId);
```

Comment:
does it make sense to call `fetchProfilesInGroupChannels` instead of this?

Reply:
You are completely right. I was using fetchMissingDirectChannelsInfo to make sure I was not missing anything, but for GMs the only thing important in this case is fetching the profiles.